### ISSUE TITLE
Angular: Getting process is not defined error

### ISSUE THREAD

**Issue by sujitpatil72** [saebekassebil/microphone-stream#60]
I am facing one issue while trying to use this package with Angular 10. Kindly help me to resolve this.

```js
import { default as MicrophoneStream } from "microphone-stream";
@Injectable({
  providedIn: "root",
})
export class AudioRecordingService {
  inputSampleRate;
  socket;
  sampleRate = 44100;
  transcribeException;
  socketError;

  startRealTimeSTT(_presignedUrl) {
    navigator.mediaDevices
      .getUserMedia({
        video: false,
        audio: true,
      })
      // ...then we convert the mic stream to binary event stream messages when the promise resolves
      .then((userMediaStream) => this.streamAudioToWebSocket(userMediaStream))
      .catch(function (error) {
        // showError('There was an error streaming your audio to Amazon Transcribe. Please try again.');
        // toggleStartStop();
      });
  }
  streamAudioToWebSocket = (userMediaStream) => {
    //let's get the mic input from the browser, via the microphone-stream module
    try {
      this.micStream = new MicrophoneStream();
    } catch (e) {
      console.log(e);
    }
  };
}
```

![image](https://user-images.githubusercontent.com/38838057/124600384-46cfb980-de84-11eb-9f83-8e8a20228aee.png)

**Comment by nfriedly** [comment]
Oof, it's expecting a pollyfill for the node.js `process` global. I think we can change that to `setTimeout` or `setImmediate`...

**Comment by nfriedly** [comment]
A couple of other notes:

1. `this.micStream = new MicrophoneStream();` this must be done in *direct* response to a click or tap in most browsers. Otherwise the audio stream will be muted. Doing it in a `.then()` callback after the `getUserMedia()` promise resolves is too late.

2.  The entire point of this library is transform the media stream API into a node.js-style one. But it sounds like you might not actually benefit from that all that much (?)

**Comment by sujitpatil72** [comment]
@nfriedly Thanks for the quick reply and suggestion. I will implement it in my code.

**Comment by sujitpatil72** [comment]
@nfriedl Hi Nathan, any updates wrt merge?

**Comment by nfriedly** [comment]
Yea, I was hoping for a review from someone, but I think it's probably fine. I'll go ahead and merge it in.

**Comment by sujitpatil72** [comment]
@nfriedly Thank you :)

**Comment by nfriedly** [comment]
Ok, the change is now published to npm in `v6.0.1`
